packages/aws-cdk/test/api/hotswap/state-machine-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,478 @@
+            StateMachineName: 'my-machine',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateMachineDefinition).toHaveBeenCalledWith({
+    definition: '{ Prop: "new-value" }',
+    stateMachineArn: 'my-machine',
+  });
+});
+
+test('calls the updateStateMachine() API when it receives only a definitionString change with Fn::Join in a state machine', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value", "AnotherProp" : "another-old-value" }',
+              ],
+            ],
+          },
+          StateMachineName: 'my-machine',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "new-value", "AnotherProp" : "another-new-value" }',
+                ],
+              ],
+            },
+            StateMachineName: 'my-machine',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateMachineDefinition).toHaveBeenCalledWith({
+    definition: '{ "Prop" : "new-value", "AnotherProp" : "another-new-value" }',
+    stateMachineArn: 'my-machine',
+  });
+});
+
+test('calls the updateStateMachine() API when it receives a change to the definitionString in a state machine that has no name', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value" }',
+              ],
+            ],
+          },
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "new-value" }',
+                ],
+              ],
+            },
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  setup.pushStackMocks(setup.stackSummaryOf('Machine', 'AWS::StepFunctions::StateMachine', 'mock-machine-resource-id'));
```

Comment:
Yep, `pushStackMocks` is definitely a weird name for this helper 🙂.

Always think of how your helpers will read in the tests.